osf/models/node.py

```diff
@@ -2125,6 +2126,12 @@ def update(self, fields, auth=None, save=True):
             elif key == 'category':
                 self.set_category(category=value, auth=auth, save=False)
             elif key == 'is_public':
+                if value is False:
+                    if self.storage_limit_status is not None:
+                        if self.is_public and self.storage_limit_status.value >= 2:
```

Comment:
It would be better to use the ENUM `OVER_PRIVATE` here instead of `2`. The comparison operator should still work.